server/src/main/java/org/apache/druid/server/metrics/DataSourceTaskIdHolder.java

```diff
@@ -46,11 +48,13 @@ public class DataSourceTaskIdHolder
   @Inject(optional = true)
   BroadcastDatasourceLoadingSpec broadcastDatasourceLoadingSpec = BroadcastDatasourceLoadingSpec.ALL;
 
+  @Nullable
```

Comment:
When can this be null?

Reply:
For peons, the `CliPeon` would [provide](https://github.com/apache/druid/blob/master/services/src/main/java/org/apache/druid/cli/CliPeon.java#L334) the values of `taskId` and `dataSource`.

For all other servers, this would be null as these fields are initialized to null::
```java
  @Named(DATA_SOURCE_BINDING)
  @Inject(optional = true)
  String dataSource = null;
  @Named(TASK_ID_BINDING)
  @Inject(optional = true)
  String taskId = null;
```